token/cli/src/command.rs

```diff
@@ -176,6 +177,51 @@ fn native_token_client_from_config(
     }
 }
 
+enum Pointer {
+    Metadata,
+    Group,
+    GroupMember,
+}
+impl Display for Pointer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(match self {
+            Pointer::Metadata => "metadata pointer",
+            Pointer::Group => "group pointer",
+            Pointer::GroupMember => "member pointer",
+        })
+    }
+}
+impl Pointer {
+    pub async fn update_address(
+        &self,
+        config: &Config<'_>,
+        token_pubkey: &Pubkey,
+        authority: &Pubkey,
+        new_address: Option<Pubkey>,
+        signing_keypairs: &BulkSigners,
+    ) -> Result<RpcClientResponse, Error> {
+        let token = token_client_from_config(config, token_pubkey, None)?;
+        match self {
+            Pointer::Metadata => {
+                token
+                    .update_metadata_address(authority, new_address, signing_keypairs)
+                    .await
+            }
+            Pointer::Group => {
+                token
+                    .update_group_address(authority, new_address, signing_keypairs)
+                    .await
+            }
+            Pointer::GroupMember => {
+                token
+                    .update_group_member_address(authority, new_address, signing_keypairs)
+                    .await
+            }
+        }
+        .map_err(|err| format!("Failed to update {}: {}", self, err).into())
```

Comment:
nit to go along with the previous
```suggestion
        .map_err(|err| format!("Failed to update {} pointer: {}", self, err).into())
```